flink-ai-flow/ai_flow_plugins/scheduler_plugins/airflow/ai_flow_operator.py

```diff
@@ -0,0 +1,112 @@
+from ai_flow.context.project_context import build_project_context
+from ai_flow.workflow.job import Job
+from ai_flow.workflow.workflow import Workflow, WorkflowPropertyKeys
+
+
+class AIFlowOperator(BaseOperator):
+    """
+    AIFlowOperator implements airflow Operator.
+    Its function is to submit and stop the job defined by the ai flow program.
+    """
+    @apply_defaults
+    def __init__(
+            self,
+            job: Job,
+            workflow: Workflow,
+            **kwargs,
+    ) -> None:
+        super().__init__(**kwargs)
+        self.job: Job = job
+        self.workflow: Workflow = workflow
+        plugins = self.workflow.properties.get(WorkflowPropertyKeys.JOB_PLUGINS)
+        module, name = plugins.get(self.job.job_config.job_type)
+        class_object = import_string('{}.{}'.format(module, name))
+        self.job_controller: JobController = class_object()
+        self.job_handle: JobHandle = None
+        self.job_runtime_env: JobRuntimeEnv = None
+
+    def context_to_job_info(self, project_name: Text, context: Any) -> JobExecutionInfo:
+        """
+        The function of this function is to turn the context of airflow into execution information of a job.
+        """
+        wi = WorkflowInfo(namespace=project_name, workflow_name=self.workflow.workflow_name)
+        we = WorkflowExecutionInfo(workflow_execution_id=str(context.get('dag_run').id),
+                                   workflow_info=wi,
+                                   start_date=str(datetime_to_int64(context.get('dag_run').start_date)),
+                                   end_date=str(datetime_to_int64(context.get('dag_run').end_date)),
+                                   status=context.get('dag_run').get_state())
+        je = JobExecutionInfo(job_name=self.job.job_name,
+                              job_execution_id=str(context.get('ti').try_number),
+                              status=context.get('ti').state,
+                              start_date=str(datetime_to_int64(context.get('ti').start_date)),
+                              end_date=str(datetime_to_int64(context.get('ti').end_date)),
+                              workflow_execution=we)
+        return je
+
+    def pre_execute(self, context: Any):
+        config = {}
+        config.update(self.workflow.properties['blob'])
+        local_repo = config.get('local_repository')
+        if local_repo is not None:
+            # Maybe Download the project code
+            if not os.path.exists(local_repo):
+                os.makedirs(local_repo)
+            blob_manager = BlobManagerFactory.get_blob_manager(config)
+            project_path: Text = blob_manager \
+                .download_project(workflow_snapshot_id=self.workflow.workflow_snapshot_id,
+                                  remote_path=self.workflow.project_uri,
+                                  local_path=local_repo)
```

Comment:
If the local path uses the `local_repository` configuration, `LocalBlobManager` doesn't need the `local_repository` configuration item.